utils/s2n_safety.h

```diff
@@ -89,9 +89,11 @@ static inline void* trace_memcpy_check(void *restrict to, const void *restrict f
     lt_check(__tmp_n, high);                    \
   } while (0)
 
-#define GUARD( x )              do {if ( (x) < 0 ) return -1;} while (0)
+#define GUARD( x )              do {if ( (x) < 0 ) return S2N_FAILURE;} while (0)
 #define GUARD_GOTO( x , label ) do {if ( (x) < 0 ) goto label;} while (0)
 #define GUARD_PTR( x )          do {if ( (x) < 0 ) return NULL;} while (0)
+#define GUARD_RETRY( x )  do {int r = (x); GUARD(r); if (S2N_AGAIN == r) return S2N_AGAIN;} while(0)
```

Comment:
not sure that GUARD_RETRY is the best name here ... this doesn't do an actual retry. Maybe GUARD_OR_AGAIN? or something better?

Reply:
How about GUARD_AGAIN for simplicity?